UMS/man/man3/reconnect.3
'\" t
.\"     Title: reconnect
.\"    Author:  SysFera SAS <info@sysfera.com>
.\" Generator: DocBook XSL Stylesheets v1.75.2 <http://docbook.sf.net/>
.\"      Date: February 2011
.\"    Manual: C++ API Reference
.\"    Source: VISHNU 1.0
.\"  Language: English
.\"
.TH "RECONNECT" "3" "February 2011" "VISHNU 1.0" "C++ API Reference"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.SH "NAME"
reconnect \- reconnects to a session that is still active
.SH "SYNOPSIS"
.HP \w'int\ vishnu::reconnect('u
.BI "int vishnu::reconnect(const\ string&\ " "userId\ " ", const\ string&\ " "password" ", const\ string&\ " "sessionId" ", Session&\ " "session" ");"
.SH "DESCRIPTION"
.PP
This command allows you to resume a session that has been opened previously and that has not yet been closed\&. You can disconnect from a session without closing it (for example if there are running commands in that session) by setting the session\*(Aqs close policy (at connection time) to CLOSE_ON_TIMEOUT\&. As sessions are linked to a specific client system, you cannot reconnect to a session that was opened on another client system\&.
.SH "ARGUMENTS"
.PP
\fIuserId \fR
.RS 4
Input argument\&. UserId represents the VISHNU user identifier\&.
.RE
.PP
\fIpassword\fR
.RS 4
Input argument\&. Password represents the password of the user\&.
.RE
.PP
\fIsessionId\fR
.RS 4
Input argument\&. SessionId is the identifier of the session defined in the database\&.
.RE
.PP
\fIsession\fR
.RS 4
Output argument\&. The session object containing session information\&.
.RE
.SH "EXCEPTIONS"
.PP
The following exceptions may be thrown:
.PP
"Internal Error: SysDS service failure" [1]
.RS 4
.RE
.PP
"Internal Error: Database returned an error" [2]
.RS 4
.RE
.PP
"Internal Error: Database connection failure" [3]
.RS 4
.RE
.PP
"Internal Error: System call failure" [4]
.RS 4
.RE
.PP
"Internal Error: Undefined exception" [9]
.RS 4
.RE
.PP
"Unknown user" [20]
.RS 4
.RE
.PP
"The user is locked" [23]
.RS 4
.RE
.PP
"The session key is unrecognized" [28]
.RS 4
.RE
.PP
"The sessionKey is expired\&. The session is closed\&." [29]
.RS 4
.RE
.PP
"The session Id is unknown" [30]
.RS 4
.RE
.PP
"The machine does not exist or it is locked" [36]
.RS 4
.RE
.SH "AUTHOR"
.PP
\fB SysFera SAS\fR <\&info@sysfera.com\&>
.RS 4
Developed the first version of these man pages as part of the VISHNU project.
.RE
.SH "COPYRIGHT"
.br
Copyright \(co 2011 SysFera SAS
.br
.PP
These manual pages are provided under the following conditions:
.PP
Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
.sp
.RS 4
.ie n \{\
\h'-04' 1.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  1." 4.2
.\}
Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
.RE
.sp
.RS 4
.ie n \{\
\h'-04' 2.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  2." 4.2
.\}
Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
.RE
.PP
This software is governed by the CECILL licence under French law and abiding by the rules of distribution of free software. You can use, modify and/ or redistribute the software under the terms of the CeCILL license as circulated by CEA, CNRS and INRIA at the following URL "http://www.cecill.info".
.PP
As a counterpart to the access to the source code and rights to copy, modify and redistribute granted by the license, users are provided only with a limited warranty and the software's author, the holder of the economic rights, and the successive licensors have only limited liability.
.PP
In this respect, the user's attention is drawn to the risks associated with loading, using, modifying and/or developing or reproducing the software by the user in light of its specific status of free software, that may mean that it is complicated to manipulate, and that also therefore means that it is reserved for developers and experienced professionals having in-depth computer knowledge. Users are therefore encouraged to load and test the software's suitability as regards their requirements in conditions enabling the security of their systems and/or data to be ensured and, more generally, to use and operate it in the same conditions as regards security.
.sp


UMS/src/cli/reconnect.cc
#include "reconnect.hh"
#include "utils.hh"
#include "connectUtils.hpp"
#include "sessionUtils.hpp"

namespace po = boost::program_options;

using namespace std;

int main (int ac, char* av[]){


	string userId;

	string password;

	string sessionId;

	string sessionKey;

	string dietConfig;



/**************** Describe options *************/


			boost::shared_ptr<Options> opt=makeConnectOptions(av[0],userId,1, dietConfig);

			opt->setPosition("userId",1);


        opt->add("sessionKey,s",
				        "The sessionKey",
						ENV,
						sessionKey);

		 opt->add("sessionId,s",
				 		 "represents the identifier of the session",
						 HIDDEN,
						 sessionId,1);

		 opt->setPosition("sessionId",1);



	try {

/**************  Parse to retrieve option values  ********************/

		opt->parse_cli(ac,av);

		opt->parse_env(env_name_mapper());

		opt->notify();





/********  Process **************************/

		if (opt->count("userId")){

			cout <<"The user identifier is " << userId << endl;

		}


		if(opt->count("sessionId")){

			cout << "The session identifier is " << sessionId <<endl;

		}


		checkVishnuConfig(*opt);

		if (opt->count("sessionKey")){

			cout <<"The session key is " << sessionKey << endl;
		}



		//Fix me
					 password=takePassword("Password: ",userId);

				cout << "The user password is " << password << endl;



/************** Call UMS connect service *******************************/

/*
               // initializing DIET
              if (diet_initialize(dietConfig.c_str(), ac, av)) {

				  cerr << "DIET initialization failed !" << endl;

				  return 1;
              }
*/

				   // get the sessionKey

							 std::string sessionFile=getSessionLocation(getppid());

							 SessionEntry session=getLastSession(sessionFile);

							 sessionKey=session.getSessionKey();

							 cout <<"sessionKey " << sessionKey <<endl;

              //reconnect(userId, password, sessionId, sessionKey);



	}// End of try bloc

	catch(po::required_option& e){

			usage(*opt,"[options] userId sessionId","required parameter is missing");

	}

	catch(std::exception& e){
		errorUsage(av[0], e.what()) ;
		return 1;
	}

	return 0;

}// end of main


